src/components/CohortBuilder/Table/ParticipantsTable.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'react-emotion';
+import { compose, withState } from 'recompose';
+import BaseDataTable from 'uikit/DataTable';
+
+const enhance = compose(withState('collapsed', 'setCollapsed', true));
+const CollapsibleMultiLineCell = enhance(({ value: data, collapsed, setCollapsed }) => {
+
+  // Display a fourth row when there is exactly 4 rows.
+  // Collapsing a single don't save any space.
+  const displayedRowCount = collapsed ? (data.length === 4 ? 4 : 3) : data.length;
+  const displayMoreButton = data.length > 4;
+  return (
+    <div>
+      {data.slice(0, displayedRowCount).map((datum, index) => (
+        <div key={index}>
+          {datum === null
+            ? '\u00A0' /* unbreakable space to avoid empty rows from collapsing in height */
+            : datum}
+        </div>
+      ))}
+      {displayMoreButton ? (
+        <div
+          onClick={() => {
+            setCollapsed(!collapsed);
+          }}
+        >
+          <div className={`showMore-wrapper ${collapsed ? 'more' : 'less'}`}>
+            {collapsed ? ' More' : 'Less'}
+          </div>
+        </div>
+      ) : null}
+    </div>
+  );
+});
+
+const StyledCollapsibleMultiLineCell = styled(CollapsibleMultiLineCell)`
```

Comment:
style might not be applied, needs to pass `className` prop down to a dom element in `CollapsibleMultiLineCell` :)

Reply:
You're right _and_ most of that is dead code, but some of it is relevant.
I'll clean that.

Reply:
which ever DOM element this is supposed to apply to, I think we'd prefer just use `styled` on that element :P the whole `className` can get complex to trace down. Sometimes it's good for re-usable components to allow overrides though :P